internal/espressoreader/espresso_helper.go

```diff
@@ -23,10 +23,12 @@ func (eh *EspressoHelper) readEspressoHeader(espressoBlockHeight uint64, url str
 	res, err := http.Get(requestURL)
 	if err != nil {
 		slog.Error("error making http request", "err", err)
+		return "" // it's not the best way to do it
```

Comment:
@ZzzzHui, I added the `return ""` to avoid the null pointer exception on line 28 at `res.Body`.

Reply:
We need to pay closer attention to error handling.